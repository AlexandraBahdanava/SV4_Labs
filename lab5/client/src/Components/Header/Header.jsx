import React, { useState } from 'react';
import { Link as RouterLink } from 'react-router-dom';
import { AppBar, Toolbar, IconButton, Typography, Drawer, List, ListItem, ListItemText } from '@mui/material';
import MenuIcon from '@mui/icons-material/Menu';
import "./Header.css"

function Header() {
    const [isMenuOpen, setIsMenuOpen] = useState(false);

    const handleMenuToggle = () => {
        setIsMenuOpen(!isMenuOpen);
    };

    return (
        <AppBar position="static">
            <Toolbar className='toolbar'>
                <Drawer
                    anchor="left"
                    open={isMenuOpen}
                    onClose={() => setIsMenuOpen(false)}
                >
                    <List>
                        <ListItem component={RouterLink} to="/" onClick={() => setIsMenuOpen(false)}>
                            <ListItemText primary="Главная" />
                        </ListItem>
                        <ListItem component={RouterLink} to="/master" onClick={() => setIsMenuOpen(false)}>
                            <ListItemText primary="Мастера" />
                        </ListItem>
                        <ListItem component={RouterLink} to="/openinghours" onClick={() => setIsMenuOpen(false)}>
                            <ListItemText primary="График работы" />
                        </ListItem>
                        <ListItem component={RouterLink} to="/appointmentform" onClick={() => setIsMenuOpen(false)}>
                            <ListItemText primary="Записаться" />
                        </ListItem>
                        <ListItem component={RouterLink} to="/record" onClick={() => setIsMenuOpen(false)}>
                            <ListItemText primary="Мои записи" />
                        </ListItem>
                    </List>
                </Drawer>
                <IconButton
                    edge="start"
                    color="inherit"
                    aria-label="menu"
                    onClick={handleMenuToggle}
                >
                    <MenuIcon />
                </IconButton>
                <Typography variant="h6" style={{ flexGrow: 1 }} className='titleHeader'>
                Учет работы мастеров маникюрного салона
                </Typography>
            </Toolbar>
        </AppBar>
    );
}

export default Header;
