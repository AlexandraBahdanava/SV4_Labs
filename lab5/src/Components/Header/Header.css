.titleHeader{
  color: aliceblue;
}

.toolbar{
  background-color: rgba(229, 124, 157, 0.852);
}