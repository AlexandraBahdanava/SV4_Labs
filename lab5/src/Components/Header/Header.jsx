import React, { useState } from 'react';
import { Link as RouterLink } from 'react-router-dom';
import { AppBar, Toolbar, IconButton, Typography, Drawer, List, ListItem, ListItemText, Link } from '@mui/material';
import MenuIcon from '@mui/icons-material/Menu';
import "./Header.css"

function Header() {
    const [isMenuOpen, setIsMenuOpen] = useState(false);

    const handleMenuToggle = () => {
        setIsMenuOpen(!isMenuOpen);
    };

    return (
        <AppBar position="static">
            <Toolbar className='toolbar'>
                <Typography variant="h6" style={{ flexGrow: 1 }} className='titleHeader'>
                Учет работы мастеров маникюрного салона
                </Typography>
               
            </Toolbar>
        </AppBar>
    );
}

export default Header;
