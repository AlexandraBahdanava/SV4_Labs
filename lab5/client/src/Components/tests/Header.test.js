import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from '../Header/Header';

describe('Header', () => {
  it('renders correctly', () => {
    const { getByText } = render(
      <Router>
        <Header />
      </Router>
    );

    expect(getByText('Учет работы мастеров маникюрного салона')).toBeTruthy();
  });

  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <Router>
          <Header />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

    beforeEach(() => {
      localStorage.clear();
    });
  
    it('renders correctly for admin role', () => {
      localStorage.setItem('role', 'admin');
      const tree = renderer
        .create(
          <Router>
            <Header />
          </Router>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  
    it('renders correctly for non-admin role', () => {
      localStorage.setItem('role', 'user');
      const tree = renderer
        .create(
          <Router>
            <Header />
          </Router>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  
    it('renders correctly when role is not set', () => {
      const tree = renderer
        .create(
          <Router>
            <Header />
          </Router>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  
    it('renders correctly when role is null', () => {
      localStorage.setItem('role', null);
      const tree = renderer
        .create(
          <Router>
            <Header />
          </Router>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });