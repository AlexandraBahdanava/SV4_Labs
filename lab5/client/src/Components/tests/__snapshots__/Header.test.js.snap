// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches the snapshot 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
>
  <div
    className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular toolbar css-hyum1k-MuiToolbar-root"
  >
    <button
      aria-label="menu"
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium css-1d6w9lk-MuiButtonBase-root-MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="MenuIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
        />
      </svg>
    </button>
    <h6
      className="MuiTypography-root MuiTypography-h6 titleHeader css-2ulfj5-MuiTypography-root"
      style={
        Object {
          "flexGrow": 1,
        }
      }
    >
      Учет работы мастеров маникюрного салона
    </h6>
  </div>
</header>
`;

exports[`Header renders correctly for admin role 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
>
  <div
    className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular toolbar css-hyum1k-MuiToolbar-root"
  >
    <button
      aria-label="menu"
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium css-1d6w9lk-MuiButtonBase-root-MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="MenuIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
        />
      </svg>
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <h6
      className="MuiTypography-root MuiTypography-h6 titleHeader css-2ulfj5-MuiTypography-root"
      style={
        Object {
          "flexGrow": 1,
        }
      }
    >
      Учет работы мастеров маникюрного салона
    </h6>
  </div>
</header>
`;

exports[`Header renders correctly for non-admin role 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
>
  <div
    className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular toolbar css-hyum1k-MuiToolbar-root"
  >
    <button
      aria-label="menu"
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium css-1d6w9lk-MuiButtonBase-root-MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="MenuIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
        />
      </svg>
    </button>
    <h6
      className="MuiTypography-root MuiTypography-h6 titleHeader css-2ulfj5-MuiTypography-root"
      style={
        Object {
          "flexGrow": 1,
        }
      }
    >
      Учет работы мастеров маникюрного салона
    </h6>
  </div>
</header>
`;

exports[`Header renders correctly when role is not set 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
>
  <div
    className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular toolbar css-hyum1k-MuiToolbar-root"
  >
    <button
      aria-label="menu"
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium css-1d6w9lk-MuiButtonBase-root-MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="MenuIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
        />
      </svg>
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <h6
      className="MuiTypography-root MuiTypography-h6 titleHeader css-2ulfj5-MuiTypography-root"
      style={
        Object {
          "flexGrow": 1,
        }
      }
    >
      Учет работы мастеров маникюрного салона
    </h6>
  </div>
</header>
`;

exports[`Header renders correctly when role is null 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
>
  <div
    className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular toolbar css-hyum1k-MuiToolbar-root"
  >
    <button
      aria-label="menu"
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium css-1d6w9lk-MuiButtonBase-root-MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="MenuIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
        />
      </svg>
    </button>
    <h6
      className="MuiTypography-root MuiTypography-h6 titleHeader css-2ulfj5-MuiTypography-root"
      style={
        Object {
          "flexGrow": 1,
        }
      }
    >
      Учет работы мастеров маникюрного салона
    </h6>
  </div>
</header>
`;
